std/algorithm/searching.d

```diff
@@ -3551,6 +3551,197 @@ if (isInputRange!Range && !isInfinite!Range &&
     assert(arr2d.maxElement!"a[1]" == arr2d[1]);
 }
 
+/**
+Iterates the passed range and selects the min and max element with `less`.
+If the extreme element occurs multiple time, the first occurrence will be
+returned.
+This function is more efficient than calling both $(LREF minElement) and
+$(LREF maxElement).
+
+Params:
+    map = custom accessor for the comparison key
+    selector = custom mapping for the extrema selection
+    minSeed = custom seed to use as initial element for min
+    maxSeed = custom seed to use as initial element for min
+    r = Range from which the extreme value will be selected
+
+Returns:
+    The extreme value according to `map` and `selector` of the passed-in values.
+
+See_Also:
+    $(LREF minElement), $(LREF maxElement)
+*/
+auto minmaxElement(alias map = "a", alias selector = "a < b", Range)(Range r)
+    if (isInputRange!Range && !isInfinite!Range)
+in
+{
+    assert(!r.empty, "r is an empty range");
+}
+body
+{
+    alias Element = ElementType!Range;
+    Unqual!Element seed = r.front;
+    r.popFront();
+    return minmaxElement!(map, selector)(r, seed, seed);
+}
+
+/// ditto
+auto minmaxElement(alias map = "a", alias selector = "a < b", Range,
+                      RangeElementType = ElementType!Range)
+                     (Range r, RangeElementType minSeed, RangeElementType maxSeed)
+    if (isInputRange!Range && !isInfinite!Range &&
+        !is(CommonType!(ElementType!Range, RangeElementType) == void))
+{
+    alias mapFun = unaryFun!map;
+    alias selectorFun = binaryFun!selector;
+
+    alias Element = ElementType!Range;
+    alias CommonElement = CommonType!(Element, RangeElementType);
+    alias MapType = Unqual!(typeof(mapFun(CommonElement.init)));
+
+    Unqual!CommonElement minElement = minSeed;
+    Unqual!CommonElement maxElement = maxSeed;
+
+    MapType minElementMapped = mapFun(minElement);
+    MapType maxElementMapped = mapFun(maxElement);
+
+    alias minmaxTuple = Tuple!(Unqual!CommonElement, "min", Unqual!CommonElement, "max");
+
+    static if (isRandomAccessRange!Range && hasLength!Range)
+    {
+        foreach (const i; 0 .. r.length)
+        {
+            MapType mapElement = mapFun(r[i]);
+            if (selectorFun(mapElement, minElementMapped))
+            {
+                minElement = r[i];
+                minElementMapped = mapElement;
+            }
+            if (selectorFun(maxElementMapped, mapElement))
+            {
+                maxElement = r[i];
+                maxElementMapped = mapElement;
+            }
```

Comment:
So here we should use the 3n/2 algorithm, even if technically slower for `<` and `int`. One great thing to do would be to specialize for this (and a few other) cases.